File: experiments/006-SmalltalkVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260915-111749-40413/workdir/src/main/java/surveycrawler/FirmSand.java
package surveycrawler;

public class FirmSand extends Ground {

    // moving

    @Override
    public void moveCrawlerTo(SurveyCrawler aCrawler, Point aNewPosition, Point aDirection) {
        aCrawler.moveTo(aNewPosition);
    }

    // facing

    @Override
    public void turnCrawlerClockwise(SurveyCrawler aCrawler) {
        aCrawler.turnFacingClockwise();
    }

    @Override
    public void turnCrawlerCounterClockwise(SurveyCrawler aCrawler) {
        aCrawler.turnFacingCounterClockwise();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-202806-59410/analysis/own-tests-project/src/surveycrawler/firm_sand.py
from surveycrawler.ground import Ground


class FirmSand(Ground):

    # moving

    def move_crawler_towards(self, a_crawler, a_direction):
        a_crawler.move_to(a_crawler.position().plus(a_direction))

    # facing

    def turn_crawler(self, a_crawler, a_turn):
        a_crawler.turn_to(a_turn)
